Mage/src/main/java/mage/abilities/common/SpellCastControllerTriggeredAbility.java

```diff
@@ -86,8 +88,13 @@ public boolean checkTrigger(GameEvent event, Game game) {
             return false;
         }
         this.getEffects().setValue("spellCast", spell);
-        if (rememberSource) {
-            this.getEffects().setTargetPointer(new FixedTarget(rememberSourceAsCard ? spell.getCard().getId() : spell.getId(), game));
+        switch (setTargetPointer) {
+            case SPELL:
+                getEffects().setTargetPointer(new FixedTarget(spell.getId(), game));
+                break;
+            case CARD:
+                getEffects().setTargetPointer(new FixedTarget(spell.getCard().getId()));
+                break;
```

Comment:
Might be better to make explicit with case NONE: break, and default: throw UnsupportedOperationException